install_sct.bat

```diff
@@ -0,0 +1,66 @@
+:: Usage: install_sct.bat <version>
+:: e.g.
+::        install_sct.bat 5.5
+
+:: Turn off command echoing for nicer output
+@echo off
+
+:: Fetch version number (should be git branch, git tag, etc.)
+set git_ref=%1
+
+:: Go to user's home directory
+pushd %HOMEPATH%
+
+:: Download SCT and check out the branch requested by the user
+echo:
+echo ### Downloading SCT source code to %HOMEPATH%...
+git clone -b %git_ref% --single-branch --depth 1 https://github.com/spinalcordtoolbox/spinalcordtoolbox.git || goto error
+
+:: Create and activate virtual environment to install SCT into
+echo:
+echo ### Using Python to create virtual environment...
+cd spinalcordtoolbox
+python -m venv venv_sct || goto error
+CALL venv_sct\Scripts\activate.bat || goto error
+echo Virtual environment created and activated successfully!
+
+:: Install SCT and its requirements
+echo:
+echo ### Installing SCT and its dependencies...
+:: Pip needs to be upgraded because default p3.7 pip won't resolve dependency conflicts correctly
+python -m pip install --upgrade pip || goto error
+pip install -r requirements.txt || goto error
+pip install -e . || goto error
+
+:: Install external dependencies
+echo:
+echo ### Downloading model files and binaries...
+FOR %%D IN (PAM50 gm_model optic_models pmj_models deepseg_sc_models deepseg_gm_models deepseg_lesion_models c2c3_disc_models binaries_win) DO sct_download_data -d %%D -k || goto error
+
+:: Give further instructions that the user add the Scripts directory to their PATH
+echo:
+echo ### Installation finished!
+echo:
+echo To use SCT's command-line scripts in Command Prompt, please follow these instructions:
+echo:
+echo 1. Open the ^Start Menu -^> Type 'path' -^> Open 'Edit the System Environment Variables'
+echo 2. Click the 'Environment variables...' button
+echo 3. Under the section 'User variables for ____', highlight the 'Path' entry, then click the 'Edit...' button.
+echo 4. Click 'New', then copy and paste this directory:
+echo:
+echo    %CD%\%venv_sct\Scripts
+echo:
+echo: 5. Click 'OK' three times. You can now access SCT's scripts in the Command Prompt.
+
+:: Return to initial directory and deactivate the virtual environment
+popd
+deactivate
+
+exit /b 0
+
+:error
+popd
+deactivate || BREAK

```

Comment:
This should also be `call deactivate`, I believe.

Reply:
I guess the no-op would be to put nothing at all, instead of `|| something`? I don't believe `cmd.exe` has an equivalent of `set -e` anyway, it just ignores errors by default.

If you want to suppress output, I would have suggested redirecting output, but I believe `call` doesn't allow pipes and input/output redirection. In any case, if I remember my DOS/Windows days, I was personally never surprised to see extra errors at the end, after a real error :stuck_out_tongue: 

It occurs to me, however, that `popd` and `call deactivate` are very likely to overwrite `%errorlevel%` in between `:error` and `echo`/`exit`. If so, maybe save it into another variable right after `:error`?

Reply:
Ahh, good catch.